internal/curves/linear.go

```diff
@@ -16,25 +17,52 @@ var (
 
 type LinearSpeedCurve struct {
 	Config configuration.CurveConfig `json:"config"`
-	Value  int                       `json:"value"`
+	Value  float64                   `json:"value"`
+}
+
+func (c *LinearSpeedCurve) Init() {
+	cfg := c.Config.Linear
+	if len(cfg.Steps) > 0 {
+		cfg.FloatSteps = make(map[int]float64)
+
+		for temp, origstr := range cfg.Steps {
+			str := strings.TrimSpace(origstr)
+			l := len(str)
+			isPercent := false
+			if l > 1 && str[l-1] == '%' {
+				isPercent = true
+				str = str[:l-1] // cut off '%' because ParseFloat() wouldn't like it
+			}
+			speed, err := strconv.ParseFloat(str, 64)
+			if err != nil {
+				ui.Warning("Invalid curve step value '%s' in %s", origstr, c.Config.ID)
```

Comment:
This should also be validated by `fan2go config validate`

Reply:
I don't know - can we?
The config code seems to be >90% autogenerated magic (and it's unclear to me what it's even generated from, [apparently it's not the `json:"bla"` annotations](https://github.com/markusressel/fan2go/pull/390#discussion_r2131072274)) and right now internal/configuration/curves.go only defines structs and has no code, so I'm not sure how to do this.

If you can give me a hint how to hook that into the curve config loading code, I'd be thankful, because I agree that it seems to be the best place to do this conversion (and validation)